.github/workflows/php-library-security.yml

```diff
@@ -0,0 +1,57 @@
+name: Security
+
+on:
+ workflow_call:
+  secrets:
+      SNYK_TOKEN:
+        required: true
+  inputs:
+    upload-sarif:
+      required: false
+      type: boolean
+      default: false
+  
+jobs:
+ snyk:
+   runs-on: ubuntu-latest
+   name: Snyk PHP Scan
+
+   steps:
+    - name: Checkout code
+      uses: actions/checkout@v3
+
+    - name: Setup PHP
+      uses: shivammathur/setup-php@v2
+      with:
+        php-version: 8.1
+        coverage: none
+        extensions: mbstring
+
+    - name: Run Snyk to check for vulnerabilities
+      uses: snyk/actions/php@master
+      env:
+        SNYK_TOKEN: ${{ secrets.SNYK_TOKEN }}
+      with:
+        args: --sarif-file-output=snyk.sarif
+
+    - name: isPrivate
+      uses: octokit/graphql-action@v2.x
+      id: isPrivate
+      with:
+          query: |
+            query release($owner:String!,$repo:String!) {
+              repository(owner:$owner,name:$repo) {
+                isPrivate
+              }
+            }
+          owner: ${{ github.repository_owner }}
+          repo: ${{ github.event.repository.name }}
+          graphqlEndpoint: ${{ github.graphql_url }}
+          token: ${{ secrets.GITHUB_TOKEN }}
+
+    - name: Upload result to GitHub Code Scanning
+      if: ${{ fromJSON(steps.isPrivate.outputs.data).repository.isPrivate  == false && inputs.upload-sarif == true  }} 
+      continue-on-error: true
+      uses: github/codeql-action/upload-sarif@v2
+      with:
+        sarif_file: snyk.sarif
```

Comment:
Do we need `security-events: write` permission for this workflow to publish to upload the sarif?